src/poprox_recommender/evaluation/evaluate.py

```diff
@@ -0,0 +1,116 @@
+import csv
+import gzip
+import json
+import logging
+
+import numpy as np
+import pandas as pd
+from lenskit.metrics import topn
+from tqdm import tqdm
+
+from poprox_concepts.api.recommendations import RecommendationRequest
+from poprox_recommender.data.mind import MindData
+from poprox_recommender.evaluation.metrics import rank_biased_overlap
+from poprox_recommender.paths import project_root
+
+logger = logging.getLogger("poprox_recommender.measurement_offline")
+
+
+def compute_rec_metric(recs_df: pd.DataFrame, request: RecommendationRequest):
+    # Make sure you run generate.py first to get the recommendations data file
+    recs = recs_df[recs_df["user"] == request.interest_profile.profile_id]
+    truth = mind_data.user_truth(request.interest_profile.profile_id)
+
+    personalized = 1 if request.interest_profile.click_history.article_ids else 0
+
+    single_rr = topn.recip_rank(recs[recs["stage"] == "final"], truth[truth["rating"] > 0])
+    single_ndcg5 = topn.ndcg(recs, truth, k=5)
+    single_ndcg10 = topn.ndcg(recs, truth, k=10)
```

Comment:
these metrics also need to filter to final.

You probabaly want to filter the recs list once (`final = recs[recs["stage"] == "final"]`).